eox_nelp/admin/certificates.py

```diff
@@ -31,34 +31,49 @@ def create_external_certificate_action(modeladmin, request, queryset):  # pylint
         request: Current django request.
         queryset: Selected records.
     """
+    errors = {}
 
     for certificate in queryset:
-        certificate_data = CertificateData(
-            user=UserData(
-                pii=UserPersonalData(
-                    username=certificate.user.username,
-                    email=certificate.user.email,
-                    name=certificate.user.profile.name,
+        try:
+            certificate_data = CertificateData(
+                user=UserData(
+                    pii=UserPersonalData(
+                        username=certificate.user.username,
+                        email=certificate.user.email,
+                        name=certificate.user.profile.name,
+                    ),
+                    id=certificate.user.id,
+                    is_active=certificate.user.is_active,
                 ),
-                id=certificate.user.id,
-                is_active=certificate.user.is_active,
-            ),
-            course=CourseData(
-                course_key=certificate.course_id,
-            ),
-            mode=certificate.mode,
-            grade=certificate.grade,
-            current_status=certificate.status,
-            download_url=certificate.download_url,
-            name=certificate.name,
-        )
-        time = certificate.modified_date.astimezone(timezone.utc)
+                course=CourseData(
+                    course_key=certificate.course_id,
+                ),
+                mode=certificate.mode,
+                grade=certificate.grade,
+                current_status=certificate.status,
+                download_url=certificate.download_url,
+                name=certificate.name,
+            )
+            time = certificate.modified_date.astimezone(timezone.utc)
 
-        create_external_certificate.delay(
-            external_certificate_data=_generate_external_certificate_data(
-                time=time,
-                certificate_data=certificate_data,
+            create_external_certificate.delay(
+                external_certificate_data=_generate_external_certificate_data(
+                    time=time,
+                    certificate_data=certificate_data,
+                )
             )
+        except Exception as exc:  # pylint: disable=broad-exception-caught
+            errors_by_class = errors.get(exc.__class__.__name__, {})
```

Comment:
This block is a little strange. What do you think about something like:
``` python
if errors.get(exc.__class__.__name__, {}):
   errors['exc.__class__.__name__']['ids'].append(certificate.id)
   errors['exc.__class__.__name__']['total'] = errors['exc.__class__.__name__']['total'] + 1
else:
   errors['exc.__class__.__name__'] = {"ids": [certificate.id], "total": 1}
```
By the way they do the same. I think so...